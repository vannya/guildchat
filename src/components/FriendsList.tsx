import React from 'react';
import { FriendRow } from './FriendRow';
import { Friend } from '../types';

export interface IFriendsListProps {
	currentChat: string;
	friends: Friend[];
	isInitialLoadingCompleted: boolean;
	onClick: (id: string) => void;
}

export const FriendsList = ({
	currentChat,
	friends,
	isInitialLoadingCompleted,
	onClick
}: IFriendsListProps): JSX.Element => {
	return (
		<div className='friends-list'>
			<h2>Friends List</h2>
			<div className='friends-list--buttons'>
				{!isInitialLoadingCompleted ? null : friends.length > 0 ? (
					friends.map(friend => {
						return (
							<FriendRow
								key={friend.id}
								id={friend.id}
								name={friend.name}
								onClick={onClick}
								activeFriend={currentChat === friend.id}
							/>
						);
					})
				) : (
					<div>Add Friends</div>
				)}
			</div>
		</div>
	);
};
