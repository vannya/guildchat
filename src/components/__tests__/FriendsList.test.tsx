import { cleanup, render } from '@testing-library/react';
import React from 'react';
import { FriendsList, IFriendsListProps } from '../FriendsList';

describe('FriendsList', () => {
	const getProps = (): IFriendsListProps => {
		return {
			currentChat: '1',
			friends: [],
			isInitialLoadingCompleted: true,
			onClick: jest.fn()
		};
	};

	it('should match snapshot', () => {
		const props = getProps();
		const { container } = render(<FriendsList {...props} />);
		expect(container).toMatchSnapshot();
		cleanup();
	});

	it('should not load if loading not completed', () => {
		const props = getProps();
		props.isInitialLoadingCompleted = false;
		const { container } = render(<FriendsList {...props} />);
		expect(container).toMatchSnapshot();
		cleanup();
	});

	it('should display friend rows if there are friends', () => {
		const props = getProps();
		props.friends = [
			{ id: '2', name: 'Friend 2' },
			{ id: '3', name: 'Friend 3' }
		];
		const { container } = render(<FriendsList {...props} />);
		const friendRows = container.querySelectorAll('.friend-row');
		expect(friendRows).toHaveLength(2);
		cleanup();
	});
});
