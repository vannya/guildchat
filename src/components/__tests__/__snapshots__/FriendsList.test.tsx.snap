// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FriendsList should match snapshot 1`] = `
<div>
  <div
    class="friends-list"
  >
    <h2>
      Friends List
    </h2>
    <div
      class="friends-list--buttons"
    >
      <div>
        Add Friends
      </div>
    </div>
  </div>
</div>
`;

exports[`FriendsList should not load if loading not completed 1`] = `
<div>
  <div
    class="friends-list"
  >
    <h2>
      Friends List
    </h2>
    <div
      class="friends-list--buttons"
    />
  </div>
</div>
`;
